pandas/io/stata.py

```diff
@@ -1865,7 +1865,7 @@ def _dtype_to_default_stata_fmt(dtype, column, dta_version=114,
         if force_strl:
             return '%9s'
     if dtype.type == np.object_:
-        inferred_dtype = infer_dtype(column.dropna())
```

Comment:
This fix is wrong IIRC.  It is important that `Empty` type is returned here for the code below to work correctly.